web/app/controllers/linkSchoolCtrl.js

```diff
@@ -4,19 +4,33 @@ var drivingSchoolCtrl = require("../controllers/drivingSchoolCtrl");
 
 // update user with schoolId for linking
 exports.linkAccToSchool = function(userId, schoolId, callback, error) {
-	User.findOneAndUpdate({
-		userId: userId
-	}, {
-		schoolId: schoolId
-	}, function(err, student) {
-		if (err) {
-			error({
-				"message": "Error finding student to link school with",
-				"error": err
+	drivingSchoolCtrl.getSchool(schoolId,
+	(results) => {
+		if(results != null){
```

Comment:
Actually, could get rid of some nesting (make Tumey proud) and do an early return
```javascript
if(results === undefined){
    error({"message": "No school found"});
    return;
}
User.findOneAndUpdate({
// continued...
```